test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -350,7 +352,7 @@ public InternalTestCluster(
             numSharedDedicatedMasterNodes, numSharedDataNodes, numSharedCoordOnlyNodes,
             autoManageMasterNodes ? "auto-managed" : "manual");
         this.nodeConfigurationSource = nodeConfigurationSource;
-        numDataPaths = random.nextInt(5) == 0 ? 2 + random.nextInt(3) : 1;
+        numDataPaths = forceSingleDataPath == false && random.nextInt(5) == 0 ? 2 + random.nextInt(3) : 1;
```

Comment:
Could we rewrite this slightly so it’s clearer: I would want it to read “if `forceSingleDataPath` is true then `1` else some random number of data paths” instead of the double negation right now. How about

```java
// use 1 data path if we are forced to, or 80% of the time that we are not, otherwise use between 2 and 4 data paths
numDataPaths = forceSingleDataPath || random.nextInt(5) != 0 ? 1 : 2 + random.nextInt(3);
```

And maybe that `random.nextInt(5) !=0` should be `random.nextDouble() < 0.8` and then it’s even clearer?

```java
// use 1 data path if we are forced to, or 80% of the time that we are not, otherwise use between 2 and 4 data paths
numDataPaths = forceSingleDataPath || random.nextDouble() < 0.8 ? 1 : 2 + random.nextInt(3);
```

And for one more iteration on clarity, how about `randomIntBetween(2, 4`)?

```java
// use 1 data path if we are forced to, or 80% of the time that we are not, otherwise use between 2 and 4 data paths
numDataPaths = forceSingleDataPath || random.nextDouble() < 0.8 ? 1 : randomIntBetween(2, 4);
```

Reply:
++, thanks, fixed in two commits 8c84446d057d186055daee98b29251ee9d1335d8 and 1ec131a1e735e367a487b6e85fdeb79a1cc34488